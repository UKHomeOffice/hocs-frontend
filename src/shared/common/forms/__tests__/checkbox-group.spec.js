import React from 'react';
import CheckboxGroup from '../checkbox-group.jsx';

const choices = [
    { label: 'isA', value: 'A' },
    { label: 'isB', value: 'B' },
    { label: 'isC', value: 'C' }
];

describe('Form checkbox group component', () => {
    it('should render with default props', () => {
        expect(
            render(<CheckboxGroup name="checkbox-group" choices={choices} updateState={() => null} />)
        ).toMatchSnapshot();
    });
    it('should render with value when passed', () => {
        expect(
            render(<CheckboxGroup name="checkbox-group" choices={choices} value={'A'} updateState={() => null} />)
        ).toMatchSnapshot();
    });
    it('should render with label when passed', () => {
        expect(
            render(<CheckboxGroup name="checkbox-group" choices={choices} label="My text field" showLabel={true} updateState={() => null} />)
        ).toMatchSnapshot();
    });
    it('should render with hint when passed', () => {
        expect(
            render(<CheckboxGroup name="checkbox-group" choices={choices} hint="Put some text in the box below" updateState={() => null} />)
        ).toMatchSnapshot();
    });
    it('should render with error when passed', () => {
        expect(
            render(<CheckboxGroup name="checkbox-group" choices={choices} error="Some error message" updateState={() => null} />)
        ).toMatchSnapshot();
    });
    it('should render disabled when passed', () => {
        expect(
            render(<CheckboxGroup name="checkbox-group" choices={choices} disabled={true} updateState={() => null} />)
        ).toMatchSnapshot();
    });
    it('should execute callback on initialization', () => {
        const mockCallback = jest.fn();
        shallow(
            <CheckboxGroup name="checkbox-group" choices={choices} updateState={mockCallback} />
        );
        expect(mockCallback).toHaveBeenCalledTimes(1);
        expect(mockCallback).toHaveBeenCalledWith({ 'checkbox-group': '' });
    });
    it('should execute callback on change', () => {
        const mockCallback = jest.fn();
        const firstValue = 'A';
        const secondValue = 'B';
        const wrapper = shallow(
            <CheckboxGroup name="checkbox-group" choices={choices} updateState={mockCallback} />
        );
        mockCallback.mockReset();

        wrapper.find(`#checkbox-group_${firstValue}`).simulate('change', { target: { value: firstValue } });
        expect(mockCallback).toHaveBeenCalledTimes(1);
        expect(mockCallback).toHaveBeenCalledWith({ 'checkbox-group': 'A' });

        wrapper.find(`#checkbox-group_${secondValue}`).simulate('change', { target: { value: secondValue } });
        expect(mockCallback).toHaveBeenCalledTimes(2);
        expect(mockCallback).toHaveBeenCalledWith({ 'checkbox-group': 'A,B' });

        wrapper.find(`#checkbox-group_${firstValue}`).simulate('change', { target: { value: firstValue } });
        expect(mockCallback).toHaveBeenCalledTimes(3);
        expect(mockCallback).toHaveBeenCalledWith({ 'checkbox-group': 'B' });
    });
});

