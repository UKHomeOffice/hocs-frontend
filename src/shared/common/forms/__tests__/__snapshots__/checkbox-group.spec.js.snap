// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form checkbox group component should render disabled when passed 1`] = `
<div
  class="govuk-form-group"
>
  <fieldset
    class="govuk-fieldset "
    disabled=""
  >
    <legend
      class="govuk-fieldset__legend"
      id="checkbox-group-legend"
    >
      <span
        class="govuk-fieldset__heading govuk-label--s"
      />
    </legend>
    <div
      class="govuk-checkboxes"
    >
      <div
        class="govuk-checkboxes__item"
      >
        <input
          class="govuk-checkboxes__input"
          id="checkbox-group_A"
          name="checkbox-group"
          type="checkbox"
          value="A"
        />
        <label
          class="govuk-label govuk-checkboxes__label"
          for="checkbox-group_A"
        >
          isA
        </label>
      </div>
      <div
        class="govuk-checkboxes__item"
      >
        <input
          class="govuk-checkboxes__input"
          id="checkbox-group_B"
          name="checkbox-group"
          type="checkbox"
          value="B"
        />
        <label
          class="govuk-label govuk-checkboxes__label"
          for="checkbox-group_B"
        >
          isB
        </label>
      </div>
      <div
        class="govuk-checkboxes__item"
      >
        <input
          class="govuk-checkboxes__input"
          id="checkbox-group_C"
          name="checkbox-group"
          type="checkbox"
          value="C"
        />
        <label
          class="govuk-label govuk-checkboxes__label"
          for="checkbox-group_C"
        >
          isC
        </label>
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`Form checkbox group component should render with default props 1`] = `
<div
  class="govuk-form-group"
>
  <fieldset
    class="govuk-fieldset "
  >
    <legend
      class="govuk-fieldset__legend"
      id="checkbox-group-legend"
    >
      <span
        class="govuk-fieldset__heading govuk-label--s"
      />
    </legend>
    <div
      class="govuk-checkboxes"
    >
      <div
        class="govuk-checkboxes__item"
      >
        <input
          class="govuk-checkboxes__input"
          id="checkbox-group_A"
          name="checkbox-group"
          type="checkbox"
          value="A"
        />
        <label
          class="govuk-label govuk-checkboxes__label"
          for="checkbox-group_A"
        >
          isA
        </label>
      </div>
      <div
        class="govuk-checkboxes__item"
      >
        <input
          class="govuk-checkboxes__input"
          id="checkbox-group_B"
          name="checkbox-group"
          type="checkbox"
          value="B"
        />
        <label
          class="govuk-label govuk-checkboxes__label"
          for="checkbox-group_B"
        >
          isB
        </label>
      </div>
      <div
        class="govuk-checkboxes__item"
      >
        <input
          class="govuk-checkboxes__input"
          id="checkbox-group_C"
          name="checkbox-group"
          type="checkbox"
          value="C"
        />
        <label
          class="govuk-label govuk-checkboxes__label"
          for="checkbox-group_C"
        >
          isC
        </label>
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`Form checkbox group component should render with error when passed 1`] = `
<div
  class="govuk-form-group govuk-form-group--error"
>
  <fieldset
    class="govuk-fieldset "
  >
    <legend
      class="govuk-fieldset__legend"
      id="checkbox-group-legend"
    >
      <span
        class="govuk-fieldset__heading govuk-label--s"
      />
    </legend>
    <span
      class="govuk-error-message"
    >
      Some error message
    </span>
    <div
      class="govuk-checkboxes"
    >
      <div
        class="govuk-checkboxes__item"
      >
        <input
          class="govuk-checkboxes__input"
          id="checkbox-group_A"
          name="checkbox-group"
          type="checkbox"
          value="A"
        />
        <label
          class="govuk-label govuk-checkboxes__label"
          for="checkbox-group_A"
        >
          isA
        </label>
      </div>
      <div
        class="govuk-checkboxes__item"
      >
        <input
          class="govuk-checkboxes__input"
          id="checkbox-group_B"
          name="checkbox-group"
          type="checkbox"
          value="B"
        />
        <label
          class="govuk-label govuk-checkboxes__label"
          for="checkbox-group_B"
        >
          isB
        </label>
      </div>
      <div
        class="govuk-checkboxes__item"
      >
        <input
          class="govuk-checkboxes__input"
          id="checkbox-group_C"
          name="checkbox-group"
          type="checkbox"
          value="C"
        />
        <label
          class="govuk-label govuk-checkboxes__label"
          for="checkbox-group_C"
        >
          isC
        </label>
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`Form checkbox group component should render with hint when passed 1`] = `
<div
  class="govuk-form-group"
>
  <fieldset
    class="govuk-fieldset "
  >
    <legend
      class="govuk-fieldset__legend"
      id="checkbox-group-legend"
    >
      <span
        class="govuk-fieldset__heading govuk-label--s"
      />
    </legend>
    <span
      class="govuk-hint"
    >
      Put some text in the box below
    </span>
    <div
      class="govuk-checkboxes"
    >
      <div
        class="govuk-checkboxes__item"
      >
        <input
          class="govuk-checkboxes__input"
          id="checkbox-group_A"
          name="checkbox-group"
          type="checkbox"
          value="A"
        />
        <label
          class="govuk-label govuk-checkboxes__label"
          for="checkbox-group_A"
        >
          isA
        </label>
      </div>
      <div
        class="govuk-checkboxes__item"
      >
        <input
          class="govuk-checkboxes__input"
          id="checkbox-group_B"
          name="checkbox-group"
          type="checkbox"
          value="B"
        />
        <label
          class="govuk-label govuk-checkboxes__label"
          for="checkbox-group_B"
        >
          isB
        </label>
      </div>
      <div
        class="govuk-checkboxes__item"
      >
        <input
          class="govuk-checkboxes__input"
          id="checkbox-group_C"
          name="checkbox-group"
          type="checkbox"
          value="C"
        />
        <label
          class="govuk-label govuk-checkboxes__label"
          for="checkbox-group_C"
        >
          isC
        </label>
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`Form checkbox group component should render with label when passed 1`] = `
<div
  class="govuk-form-group"
>
  <fieldset
    class="govuk-fieldset "
  >
    <legend
      class="govuk-fieldset__legend"
      id="checkbox-group-legend"
    >
      <span
        class="govuk-fieldset__heading govuk-label--s"
      >
        My text field
      </span>
    </legend>
    <div
      class="govuk-checkboxes"
    >
      <div
        class="govuk-checkboxes__item"
      >
        <input
          class="govuk-checkboxes__input"
          id="checkbox-group_A"
          name="checkbox-group"
          type="checkbox"
          value="A"
        />
        <label
          class="govuk-label govuk-checkboxes__label"
          for="checkbox-group_A"
        >
          isA
        </label>
      </div>
      <div
        class="govuk-checkboxes__item"
      >
        <input
          class="govuk-checkboxes__input"
          id="checkbox-group_B"
          name="checkbox-group"
          type="checkbox"
          value="B"
        />
        <label
          class="govuk-label govuk-checkboxes__label"
          for="checkbox-group_B"
        >
          isB
        </label>
      </div>
      <div
        class="govuk-checkboxes__item"
      >
        <input
          class="govuk-checkboxes__input"
          id="checkbox-group_C"
          name="checkbox-group"
          type="checkbox"
          value="C"
        />
        <label
          class="govuk-label govuk-checkboxes__label"
          for="checkbox-group_C"
        >
          isC
        </label>
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`Form checkbox group component should render with value when passed 1`] = `
<div
  class="govuk-form-group"
>
  <fieldset
    class="govuk-fieldset "
  >
    <legend
      class="govuk-fieldset__legend"
      id="checkbox-group-legend"
    >
      <span
        class="govuk-fieldset__heading govuk-label--s"
      />
    </legend>
    <div
      class="govuk-checkboxes"
    >
      <div
        class="govuk-checkboxes__item"
      >
        <input
          class="govuk-checkboxes__input"
          id="checkbox-group_A"
          name="checkbox-group"
          type="checkbox"
          value="A"
        />
        <label
          class="govuk-label govuk-checkboxes__label"
          for="checkbox-group_A"
        >
          isA
        </label>
      </div>
      <div
        class="govuk-checkboxes__item"
      >
        <input
          class="govuk-checkboxes__input"
          id="checkbox-group_B"
          name="checkbox-group"
          type="checkbox"
          value="B"
        />
        <label
          class="govuk-label govuk-checkboxes__label"
          for="checkbox-group_B"
        >
          isB
        </label>
      </div>
      <div
        class="govuk-checkboxes__item"
      >
        <input
          class="govuk-checkboxes__input"
          id="checkbox-group_C"
          name="checkbox-group"
          type="checkbox"
          value="C"
        />
        <label
          class="govuk-label govuk-checkboxes__label"
          for="checkbox-group_C"
        >
          isC
        </label>
      </div>
    </div>
  </fieldset>
</div>
`;
